exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsAggBatch.java

```diff
@@ -0,0 +1,256 @@
+          + collector.toErrorString());
+    }
+  }
+
+  private StreamingAggregator codegenAggregator(List<LogicalExpression> keyExprs,
+      List<LogicalExpression> valueExprs, List<TypedFieldId> keyOutputIds)
+          throws SchemaChangeException, ClassTransformationException, IOException {
+    ClassGenerator<StreamingAggregator> cg =
+        CodeGenerator.getRoot(StreamingAggTemplate.TEMPLATE_DEFINITION, context.getFunctionRegistry(),
+            context.getOptions());
+
+    LogicalExpression[] keyExprsArray = new LogicalExpression[keyExprs.size()];
+    LogicalExpression[] valueExprsArray = new LogicalExpression[valueExprs.size()];
+    TypedFieldId[] keyOutputIdsArray = new TypedFieldId[keyOutputIds.size()];
+
+    keyExprs.toArray(keyExprsArray);
+    valueExprs.toArray(valueExprsArray);
+    keyOutputIds.toArray(keyOutputIdsArray);
+
+    setupIsSame(cg, keyExprsArray);
+    setupIsSameApart(cg, keyExprsArray);
+    addRecordValues(cg, valueExprsArray);
+    outputRecordKeys(cg, keyOutputIdsArray, keyExprsArray);
+    outputRecordKeysPrev(cg, keyOutputIdsArray, keyExprsArray);
+
+    cg.getBlock("resetValues")._return(JExpr.TRUE);
+    getIndex(cg);
+
+    container.buildSchema(SelectionVectorMode.NONE);
+    StreamingAggregator agg = context.getImplementationClass(cg);
+    agg.setup(oContext, incoming, this);
+    return agg;
+  }
+
+  private boolean isImplicitFileColumn(MaterializedField mf) {
+    return ImplicitColumnExplorer.initImplicitFileColumns(context.getOptions()).get(mf.getName()) != null;
+  }
+
+  protected StreamingAggregator createAggregatorInternal()
+      throws SchemaChangeException, ClassTransformationException, IOException {
+    container.clear();
+
+    List<LogicalExpression> keyExprs = Lists.newArrayList();
+    List<LogicalExpression> valueExprs = Lists.newArrayList();
+    List<TypedFieldId> keyOutputIds = Lists.newArrayList();
+    GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+    calendar.setTimeInMillis(System.currentTimeMillis());
+
+    if (this.getPopConfig() instanceof StatisticsAggregate
+        && (((StatisticsAggregate) this.getPopConfig()).getPhase() == OperatorPhase.PHASE_1of1
+            || ((StatisticsAggregate) this.getPopConfig()).getPhase() == OperatorPhase.PHASE_2of2)) {
+      createKeyColumn("schema",
+          ValueExpressions.getBigInt(schema++),
+          keyExprs,
+          keyOutputIds
+      );
+      createKeyColumn("computed",
+          ValueExpressions.getDate(calendar),
+          keyExprs,
+          keyOutputIds
+      );
+    }
+
+    MapVector cparent = new MapVector("column", oContext.getAllocator(), null);
+    container.add(cparent);
+    for (MaterializedField mf : incoming.getSchema()) {
+      // Ignore implicit columns
+      if (!isImplicitFileColumn(mf)) {
+        createNestedKeyColumn(
+            cparent,
+            mf.getLastName(),
```

Comment:
Is this a bug? Suppose my incoming is JSON:

```
{ a: { x: "foo", y: "bar" }, b: { x: 10, y: 20 } }
```

(Indeed, this is the very structure that this operator creates!)

If we use only the last name, then under our "column" map we try to create two "x"s and two "y"s. Does the last one win? Does that cause a problem later as we try to store values from two columns into a single summary column?